packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -252,6 +252,47 @@ Future<XcodeBuildResult> buildXcodeProject({
     );
   }
 
+  // If buildNumber is not specified, keep the project untouched.
+  if (buildInfo.buildNumber != null) {
+    final Status buildNumberStatus =
+    logger.startProgress('Running agvtool new-version...', expectSlowOperation: true);
+    final RunResult buildNumberResult = await runAsync(
```

Comment:
Wrap this in a `try/finally` where you call `buildNumberStatus.stop()` in the `finally` block. (the `exitCode` check can just move into the `try` block)